k8s/object.bzl

```diff
@@ -96,9 +96,67 @@ EOF""".format(
     ctx.outputs.yaml,
   ] + all_inputs))
 
+def _common_impl(ctx, template):
+  kwargs = {
+      "cluster": ctx.attr.cluster,
+      "namespace": ctx.attr.namespace,
+      "script": ctx.outputs.executable.path,
+      "kind": ctx.attr.kind,
+  }
+  files = [ctx.executable._resolver]
+
+  if hasattr(ctx.executable, "resolved"):
+    kwargs["resolve_script"] = ctx.executable.resolved.short_path
+    files += [ctx.executable.resolved]
+    files += list(ctx.attr.resolved.default_runfiles.files)
+
+  if hasattr(ctx.files, "unresolved"):
+    kwargs["unresolved"] = ctx.files.unresolved[0].short_path
+    files += ctx.files.unresolved
+
+  ctx.action(
+      command = template.format(**kwargs),
+      outputs = [ctx.outputs.executable],
+  )
+  return struct(runfiles = ctx.runfiles(files = files))
+
+def _create_impl(ctx):
+  return _common_impl(ctx, """cat > {script} <<"EOF"
+#!/bin/bash -e
+set -o pipefail
+./{resolve_script} | \
+  kubectl --cluster="{cluster}" --namespace="{namespace}" create -f -
+EOF""")
+
+def _replace_impl(ctx):
+  return _common_impl(ctx, """cat > {script} <<"EOF"
+#!/bin/bash -e
+set -o pipefail
+./{resolve_script} | \
+  kubectl --cluster="{cluster}" --namespace="{namespace}" replace -f -
+EOF""")
+
+def _describe_impl(ctx):
+  return _common_impl(ctx, """cat > {script} <<"EOF"
+#!/bin/bash -e
+set -o pipefail
+kubectl --cluster="{cluster}" --namespace="{namespace}" describe {kind} \
+  "$(kubectl create --dry-run -f "{unresolved}" | cut -d'"' -f 2)"
```

Comment:
What's the create --dry-run for?

Reply:
I hoisted this into a variable name.  Essentially I'm abusing this to extract the resource name.  I'm open to suggestions :-/